### DISCUSSION TITLE
Need help on generating large number of grains with mesh

### DISCUSSION THREAD

**Discussion by ashokambig** [neperfepx/neper#588] [Q&A] [answered]
Dear Neper members,
I need a help on creating a domain with at least 25000 grains in a 3D domain.
This is for a micro structure simulation at nanometer level  using abaqus.
Size of the domain can be anything. With such large grains number we can expect large number of elements.
Therefore I would like to restrict number of elements to 20 millions.
Shape of the grains can be of standard voronoi type, but I would like to have almost same shape and sphericity of grains.

Summary of my problem:
**Domain : can be anything (e.g cube with sides (1000,1000,1000))**
**Tesselation type  : Standard Voronoi, almost equal sized grain (shape and sphericity almost same)** 
**Number of grains : ~ 25000**
**Maximum number of finite elements ~ 20 millions (can be a bit higher too)**

I  have tried below option. This works for 6000 grains. But  I dont know how I can improve it to generate higher number of grains.
Can anyone help me?

************************************************
export OMP_NUM_THREADS=1 
neper --rcfile none -T -n 6000 -domain "cube(0.02,0.015,0.015)" -morpho gg -o abq_input
neper --rcfile none -M abq_input.tess -elttype tet -order 2 -rcl 0.5 -faset none -format inp 

Regards
Ashok

**Comment by rquey** [comment, marked as answer]
I may be missing something, but you are getting 6000 cells simply because you are using `-n 6000`.  If you want more cells, use a larger value.
